recipes/hycom-glby008-expt-93/recipe.py

```diff
@@ -0,0 +1,44 @@
+import pandas as pd
+
+from pangeo_forge_recipes.patterns import ConcatDim, FilePattern, MergeDim
+from pangeo_forge_recipes.recipes import XarrayZarrRecipe
+
+
+# Data is spread across three NetCDF files
+components = ["ssh", "ts3z", "uv3z"]
+
+
+def make_filename(component, time):
+    yyyy = time.strftime("%Y")
+    yyyymmdd = time.strftime("%Y%m%d")
+
+    return (
+        "http://data.hycom.org/datasets/GLBy0.08/expt_93.0/data/hindcasts/"
+        f"{yyyy}/hycom_glby_930_{yyyymmdd}12_t000_{component}.nc"
+    )
+
+
+dates = pd.date_range("20181204", "20220301", freq="D")
+
+time_concat_dim = ConcatDim("time", dates, nitems_per_file=1)
+merge_dim = MergeDim(name="component", keys=components)
+pattern = FilePattern(make_filename, time_concat_dim, merge_dim)
+recipe = XarrayZarrRecipe(
+    pattern,
+    target_chunks={"depth": 1, "lat": 500, "lon": 500},
+    xarray_open_kwargs={"drop_variables": ["tau"]},  # Tau uses a non-standard calendar
+)
+
+
+def dap_format_fn(time):
+    # Need a concat dim even if we will ignore it...
+    return "http://tds.hycom.org/thredds/dodsC/GLBy0.08/expt_93.0"
+
+
+dap_pattern = FilePattern(dap_format_fn, ConcatDim("time", ["1"]), is_opendap=True)
+dap_recipe = XarrayZarrRecipe(
+    dap_pattern,
+    subset_inputs={"time": 1, "depth": 1, "lon": 500, "lat": 500},
```

Comment:
This line does not look right to me. Compare to the tutorial - https://pangeo-forge.readthedocs.io/en/latest/tutorials/xarray_zarr/opendap_subset_recipe.html#define-the-recipe 

What is your goal with specifying subset inputs? You should probably have something like `{'time': N}` where `N` is the number of pieces you want to split it into.

Reply:
Oops. I was thinking in chunk sizes here.

Reply:
This may explain why the Prefect Cloud worker died when trying to run the `-opendap` recipe: it may have tried to load the entire temporal extent of the data into memory at once.

It's perhaps a bit counterintuitive that `target_chunks` takes a dict `{'dim': N}` where `N` is the _length (i.e. number of indices)_ of the specified dimension to include in a single chunk, whereas `subset_inputs` also takes `{'dim': N}` but here `N` is interpreted as the _number of subsets_ for the dimension.